### ISSUE TITLE
`wilcoxon()` provides two (different) p-values when `exact = TRUE`

### ISSUE THREAD

**Issue by cwolock** [statdivlab/rigr#30]
As currently written, `wilcoxon()` is (as far as I can tell) largely a copy-pasted version of `wilcox.test()` from the stats package. That copied code provides a p-value (variable `PVAL` in the code), either using an exact test, or Normal approximation. 

One of the additions in `wilcoxon()` is the `inf` return object, which is described as "a formatted table of inference values, for printing." This object provides the `PVAL` mentioned above, but also includes the Z-score and corresponding Normal approximation p-value. This p-value will correspond to `PVAL` when `exact = FALSE`. However, when `exact = TRUE` is used, `PVAL` is an exact p-value, but the Normal approximation p-value is still included in `inf`. 

Long story short: when `exact = TRUE`, `wilcoxon()` provides both the exact and approximate p-value, which I think would be confusing to a user. To see this run:

`set.seed(2)`
`y <- rnorm(100)`
`wilcoxon(y, exact = TRUE)`

The `inf` return object has p-value 0.70012 (exact) and 0.69762 (Normal approximation). 

My feeling is that we shouldn't report two different p-values. Thoughts?

**Comment by bdwilliamson** [comment]
Seems to me that we should print/report the exact p-value when `exact = TRUE` and print/report the approximate p-value when `exact = FALSE`, but should be fine to have both p-values in the object that is returned by `wilcoxon`.

**Comment by cwolock** [comment]
Right, I agree. I guess I don't understand the exact purpose of the `inf` return object (which is printed when the test is run). Do you have a sense of why this was added?

**Comment by cwolock** [comment]
I guess it adds the Z-score to the output. But do we want to provide a Z-score when `exact = TRUE`?

**Comment by bdwilliamson** [comment]
It seems to me that most of the copy/pasting was to use internal objects from wilcox.test for nice printing (e.g., computing observed and expected values on line 423). It's possible that this could be done better by running wilcox.test to compute the test statistic etc. and computing these objects by hand.

I'm pretty sure we were trying to replicate the output from Stata's wilcoxon function (see output [here](https://www.statology.org/wilcoxon-signed-rank-test-stata/)) with this, so the `inf` object collects information for the hypothesis testing portion?

**Comment by bdwilliamson** [comment]
Honestly, a lot of the printing (and hence a lot of the internal work within `uwIntroStats`) was meant to match Stata output. Since this isn't the primary goal of the package anymore, we could probably ditch some of these objects.

(and to answer your other question, I don't think we should provide a Z-score if `exact = TRUE`)

**Comment by cwolock** [comment]
Interesting re the Stata stuff - that's definitely good to know. It looks like Stata does provide both an exact p-value and approximate p-value, so that makes sense now. 

I think one possible reason for my confusion here is that the `print.wilcoxon` method is not exported. So if `my.wilcox` is a `wilcoxon` object, `print(my.wilcox)` just prints the object without any of the desired formatting. But the extra objects make a lot more sense when that method is used. We should probably change that to be exported.

**Comment by cwolock** [comment]
This bug was fixed by da57003e467970a52a1123a01c411df82c25c973
